src/JPEGView/PNGWrapper.cpp

```diff
@@ -190,6 +179,19 @@ bool BeginReading(void* buffer, size_t sizebytes, bool& outOfMemory)
 		// skip png signature since we already checked it
 		png_set_sig_bytes(png_ptr, 8);
 		// custom read function so we can read from memory
+		png_rw_ptr read_data_fn = [](png_structp png_ptr, png_bytep outbuffer, png_size_t sizebytes)
```

Comment:
just out of curiosity, this is the same chunk of code block that is the removed method, and embedded in this method... 

Is there a design choice to put the method inside a method like this vs having it on the outside as a separate method?

Reply:
It would have to be a class method since it needs access to the cache, which would mean including png.h in PNGWrapper.h. I wanted to avoid this so that ImageLoadThread.cpp doesn't have access to everything in libpng. Also it should never be directly called from anywhere anyway.

Reply:
thanks for the explanation, helps a ton!  Right, libpng methods aren't scoped that well.